### ISSUE TITLE
Segment Dragging will out of waveform.

### ISSUE THREAD

**Issue by jason30704** [bbc/peaks.js#547]
Hello,
When enableSegmentDragging(true) is set, segment will not detect whether endTime is out of range.

https://github.com/user-attachments/assets/15087faf-98e3-427a-a49d-45e58f94c0c7

**Comment by jason30704** [comment]
I'm not sure if this is an issue or not.
I can achieve almost complete functionality by trying the method below.

![image](https://github.com/user-attachments/assets/32fb6dec-2964-4fa5-a60f-de05cffc6914)

However, overlay does not stop perfectly like startMarker and endMarker.

https://github.com/user-attachments/assets/089d9e47-e05a-4026-a1c0-683bb80640f6

**Comment by chrisn** [comment]
Yes, it's inconsistent. You can drag a segment so that it ends after the right edge of the waveform view, but dragging the segment marker is limited to the width of the waveform view.

**Comment by jason30704** [comment]
From a user's perspective, I think it's strange that a segment would exceed the scope of a wave.
In practice, there really isn’t much of a problem.

**Comment by chrisn** [comment]
So, there are three separate changes we could make:
1. Allow point and segment handles to be dragged beyond the limit of the waveform view.
2. Limit point and segment dragging to within the waveform duration.
3. Limit point and segment times to within the waveform duration when calling `peaks.points.add()`, `peaks.segments.add()`, `peaks.points.update()`, and `peaks.segments.update()`.

I'm hesitant about 2 and 3.
